File: examples-sdk/python/008-basic-credit-quality-monitor.py
#!/usr/bin/env python3
"""008-basic-credit-quality-monitor — Python SDK equivalent
Compile: vil compile --from python --input 008-basic-credit-quality-monitor.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

pipeline = VilPipeline("CreditQualityMonitorPipeline", 3082)
pipeline.sink(port=3082, path="/quality-check", name="quality_monitor_sink")
pipeline.source(format="json", name="quality_credit_source")
pipeline.route("sink.trigger_out", "source.trigger_in", "LoanWrite")
pipeline.route("source.response_data_out", "sink.response_data_in", "LoanWrite")
pipeline.route("source.response_ctrl_out", "sink.response_ctrl_in", "Copy")
pipeline.compile()

File: examples-sdk/typescript/008-basic-credit-quality-monitor.ts
#!/usr/bin/env tsx
// 008-basic-credit-quality-monitor — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 008-basic-credit-quality-monitor.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const p = new VilPipeline("CreditQualityMonitorPipeline", 3082);
p.sink({ port: 3082, path: "/quality-check", name: "quality_monitor_sink" });
p.source({ name: "quality_credit_source", format: "json" });
p.route("sink.trigger_out", "source.trigger_in", "LoanWrite");
p.route("source.response_data_out", "sink.response_data_in", "LoanWrite");
p.route("source.response_ctrl_out", "sink.response_ctrl_in", "Copy");
p.compile();
